definition_validator/validator.cc

```diff
@@ -503,12 +503,26 @@ void validateOverriding(const core::Context ctx, const ast::ExpressionPtr &tree,
     if (klassData->superClass().exists()) {
         auto superMethod = klassData->superClass().data(ctx)->findMethodTransitive(ctx, name);
         if (superMethod.exists()) {
+            if (superMethod.data(ctx)->flags.isOverloaded) {
+                ENFORCE(!superMethod.data(ctx)->name.isOverload(ctx));
+                auto overload = ctx.state.lookupNameUnique(core::UniqueNameKind::Overload, name, 1);
+                superMethod = superMethod.data(ctx)->owner.data(ctx)->findMethod(ctx, overload);
+                ENFORCE(superMethod.exists());
+            }
+
             overriddenMethods.emplace_back(superMethod);
         }
     }
     for (const auto &mixin : klassData->mixins()) {
         auto superMethod = mixin.data(ctx)->findMethod(ctx, name);
         if (superMethod.exists()) {
+            if (superMethod.data(ctx)->flags.isOverloaded) {
+                ENFORCE(!superMethod.data(ctx)->name.isOverload(ctx));
+                auto overload = ctx.state.lookupNameUnique(core::UniqueNameKind::Overload, name, 1);
+                superMethod = superMethod.data(ctx)->owner.data(ctx)->findMethod(ctx, overload);
+                ENFORCE(superMethod.exists());
+            }
+
```

Comment:
I'm curious about these two stanzas. Later on, when we read from the overridenMethods that we're populating here, we skip over overloaded methods (see the top of `validateCompatibleOverride`). So the only reason to keep this logic would be to deal with like declaring `override` on the overload methods?

But it seems like even if we're trying to keep the existing behavior, we're not handling things properly, because e.g. you could have

```ruby
class Parent
  sig { abstract.void }
  def foo; end
end

class Child < Parent
  sig { override.void }
  sig { params(x: Integer).void }
  def foo; end
end
```

and Sorbet would say "yep this is fine"

but it seems like the actual behavior is that we just don't do any of that?

[example on sorbet.run →](https://sorbet.run/#%23%20typed%3A%20__STDLIB_INTERNAL%0Aclass%20Module%3B%20include%20T%3A%3ASig%3B%20end%0A%0Aclass%20Parent%0A%20%20extend%20T%3A%3AHelpers%0A%20%20abstract!%0A%0A%20%20sig%20%7B%20abstract.void%20%7D%0A%20%20def%20foo%3B%20end%0Aend%0A%0Aclass%20Child1%20%3C%20Parent%0A%20%20sig%20%7B%20override.void%20%7D%0A%20%20sig%20%7B%20params%28x%3A%20Integer%29.void%20%7D%0A%20%20def%20foo%28x%29%3B%20end%0Aend%0A%0Aclass%20Child2%20%3C%20Parent%0A%20%20sig%20%7B%20void%20%7D%0A%20%20sig%20%7B%20override.params%28x%3A%20Integer%29.void%20%7D%0A%20%20def%20foo%28x%29%3B%20end%0Aend)

So I'm kind of curious whether we should even clutter this code up with code to attempt to handle overloads if we're not actually fixing the core issues (or, if we do want to fix the core issues, that's great too, either in this change or in a separate change, with tests).

Reply:
We could avoid the clutter by migrating information about abstractness back from the first sig to the `<U ...>` named symbol, which would be a nice cleanup. Currently the only information propagated to the `<U ...>` named symbol is whether or not overloads are present, so if we don't look at the first sig as well we might miss that it's actually abstract as well.

Reply:
Ah, if we punt on overloads we'll stop checking that `Enumerable#each` is defined when you `include Enumerable`, so I'm not sure we can skip them.

Reply:
ugh

Reply:
ok in that case let's do whatever we need to do to at least get it to be no worse than what exists now, and we can fix this all separately